implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalFxMeterRegistry.java

```diff
@@ -216,4 +243,30 @@ protected void publish() {
     protected TimeUnit getBaseTimeUnit() {
         return TimeUnit.SECONDS;
     }
+
+    private DistributionStatisticConfig updateConfig(
+            DistributionStatisticConfig distributionStatisticConfig) {
+        // Add the +Inf bucket since the "count" resets every export.
+        double[] sla = distributionStatisticConfig.getServiceLevelObjectiveBoundaries();
+        if (sla == null || sla.length == 0) {
+            return distributionStatisticConfig;
+        }
+        double[] newSLA = sla;
+        if (!isPositiveInf(sla[sla.length - 1])) {
+            newSLA = Arrays.copyOf(sla, sla.length + 1);
+            newSLA[newSLA.length - 1] = Double.MAX_VALUE;
+        }
+        return DistributionStatisticConfig.builder()
+                // Set the expiration duration for the histogram counts to be effectively a lifetime.
+                // Without this, the counts are reset every expiry duration.
+                .expiry(Duration.ofNanos(Long.MAX_VALUE)) // effectively a lifetime
```

Comment:
This has an effect on `max`:
Timers usually use `TimeWindowMax` to produce a _decaying_ max value. This change disables this decaying behavior and a `max` value that shows the highest recorder value since application has started is not very useful.

Reply:
I understand, but this is a problem for the histogram buckets. There is not an easy way to have both of these to reset when "report" so, I am going with the least problematic solution which indeed makes max useless.

@mateuszrzeszutek Do you think we should remove the "max" metric in this case since it is useless?

Reply:
`DistributionSummary` uses `TimeWindowMax` too, right?

> @mateuszrzeszutek Do you think we should remove the "max" metric in this case since it is useless?

Why not, it is useless after all. I don't really have a strong opinion about it though, if there's any reason for leaving it as it is (convention?) then I'm fine with keeping it too.